src/main/java/fi/metatavu/metaform/server/files/FileUploadServlet.java

```diff
@@ -38,6 +39,36 @@
   @Inject
   private FileController fileController;
   
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String fileRef = req.getParameter("fileRef");
+    if (StringUtils.isBlank(fileRef)) {
+      resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    
+    File fileData = fileController.getFileData(fileRef);
+    if (fileData == null) {
+      resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    
+    resp.setContentType(fileData.getMeta().getContentType());
+    
+    try {
+      ServletOutputStream servletOutputStream = resp.getOutputStream();
+      try {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Extract this nested try block into a separate method. ([squid:S1141](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1141))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=57&issues=AWWZ3fCn7khxiKmGi_oR&open=AWWZ3fCn7khxiKmGi_oR)

[ISSUE_KEY]: AWWZ3fCn7khxiKmGi_oR